lib/Net/Async/Blockchain/Client/Websocket.pm

```diff
@@ -101,8 +101,12 @@ sub websocket_client : method {
                     my (undef, $frame) = @_;
                     $self->source->emit(decode_json_utf8($frame));
                 },
+                on_closed => sub {
+                    die "Connection closed by peer";
+                },
```

Comment:
What will happen when this `die`s? Will the service get restarted?

Reply:
It will be decided by the module using it, but we should not reach this point anymore, is that why I added the die here, basically if we reach this is because the node is down, we don't close the connection in any point in the code, it needs to stay alive.

Reply:
Should we not implement reconnect (x times) here? The server could close the connection for any reason. Our own websocket API closes whenever we do a release for example.